setup/swc-installation-test-2.py

```diff
@@ -215,15 +227,23 @@ def __init__(self, checker, message, causes=None):
     def get_url(self):
         system = _platform.system()
         version = None
-        for pversion in (
-            'linux_distribution',
-            'mac_ver',
-            'win32_ver',
-            ):
-            value = getattr(_platform, pversion)()
-            if value[0]:
-                version = value[0]
-                break
+        # This is less compact than the original setting, but easier to read
+        # And I don't know why we need to check the linux distribution if
+        # we already established that the system is linux
```

Comment:
Where has been established that it's Linux already? This bits are to obtain the version of the systems. So they can tell us more info about their system.

Regarding the readability of that code, the above is DRY-er (don't repeat yourself) while being clear enough (I'd use a dictionary instead a for loop though). I suppose that in this example there's not much new systems that will be required to be added to that list, so your rewrite is good too.

Reply:
system = _platform.system(), just a few lines above,  returns "Linux", "Darwin" or "Windows". So why would you need to check linux_distribution if that line already tells you it's a Mac or Windows device?

Reply:
It's to know the version. So you can see whether you've got an ubuntu or fedora, or different versions of Mac or Windows.

Reply:
Right, but you need to call platform.linux_distribution (or did) to get that for Linux, and platform.mac_ver to get that if the system returned Mac. What I think you are suggesting is have a three-item dictionary where the key would be the output of _platform.system() and the value would be the attribute in either platform or distro and then it calls it accordingly rather than going with the if statements?

Reply:
I've added this commit [41d2518](https://github.com/UCL/2023-03-09-UCL-DasH/pull/6/commits/41d2518cec6be8772b4b1bac476dad85449a9141) with that change.